app/chat/ls.go

```diff
@@ -334,3 +337,207 @@ func applyPeers(ctx context.Context, manager *peers.Manager, entities peer.Entit
 
 	return manager.Apply(ctx, users, chats)
 }
+
+// fetchDialogsWithErrorHandling manually iterates through dialogs using the raw Telegram API
+// to gracefully handle errors from problematic dialogs (deleted/inaccessible channels).
+// Instead of failing completely when ExtractPeer fails in gotd's iterator, it logs errors
+// and continues, skipping bad dialogs.
+func fetchDialogsWithErrorHandling(ctx context.Context, api *tg.Client, log *zap.Logger) ([]dialogs.Elem, int) {
+	const batchSize = 100
+	var (
+		allElems   []dialogs.Elem
+		skipped    int
+		offsetID   int
+		offsetDate int
+		offsetPeer tg.InputPeerClass = &tg.InputPeerEmpty{}
+	)
+
+	for {
+		// Fetch a batch of dialogs using raw API
+		result, err := api.MessagesGetDialogs(ctx, &tg.MessagesGetDialogsRequest{
+			OffsetDate: offsetDate,
+			OffsetID:   offsetID,
+			OffsetPeer: offsetPeer,
+			Limit:      batchSize,
+		})
+		if err != nil {
+			log.Error("failed to fetch dialog batch", zap.Error(err))
+			break
+		}
+
+		var (
+			dialogsSlice []tg.DialogClass
+			messages     []tg.MessageClass
+			users        []tg.UserClass
+			chats        []tg.ChatClass
+		)
+
+		switch d := result.(type) {
+		case *tg.MessagesDialogs:
+			dialogsSlice = d.Dialogs
+			messages = d.Messages
+			users = d.Users
+			chats = d.Chats
+		case *tg.MessagesDialogsSlice:
+			dialogsSlice = d.Dialogs
+			messages = d.Messages
+			users = d.Users
+			chats = d.Chats
+		case *tg.MessagesDialogsNotModified:
+			// No more dialogs
+			return allElems, skipped
+		default:
+			log.Error("unexpected dialog type", zap.String("type", fmt.Sprintf("%T", result)))
+			return allElems, skipped
+		}
+
+		if len(dialogsSlice) == 0 {
+			break
+		}
+
+		// Build entities map for this batch
+		// Convert slices to maps as required by peer.NewEntities
+		userMap := make(map[int64]*tg.User)
+		for _, u := range users {
+			if user, ok := u.(*tg.User); ok {
+				userMap[user.ID] = user
+			}
+		}
+
+		chatMap := make(map[int64]*tg.Chat)
+		channelMap := make(map[int64]*tg.Channel)
+		for _, c := range chats {
+			switch chat := c.(type) {
+			case *tg.Chat:
+				chatMap[chat.ID] = chat
+			case *tg.Channel:
+				channelMap[chat.ID] = chat
+			}
+		}
+
+		entities := peer.NewEntities(userMap, chatMap, channelMap)
+
+		// Process each dialog in this batch
+		for _, d := range dialogsSlice {
+			dialog, ok := d.(*tg.Dialog)
+			if !ok {
+				continue
+			}
+
+			// Find the peer ID for logging purposes
+			var peerID int64
+			switch p := dialog.Peer.(type) {
+			case *tg.PeerUser:
+				peerID = p.UserID
+			case *tg.PeerChat:
+				peerID = p.ChatID
+			case *tg.PeerChannel:
+				peerID = p.ChannelID
+			default:
+				log.Warn("unknown peer type", zap.String("type", fmt.Sprintf("%T", p)))
+				skipped++
+				continue
+			}
+
+			// Try to extract the peer - THIS IS WHERE THE ORIGINAL ERROR OCCURS
```

Comment:
Please link to the original issue here. This helps developers track the context in future working.